### PR TITLE
[MISC] Rename {core->utlity}/char_operations/predicate.hpp.

### PR DESCRIPTION
**Pull request by smehringer**
Part of https://github.com/seqan/product_backlog/issues/160

Header File: {core->utlity}/char_operations/predicate.hpp.

Note this is only one header file, since it already touches a lot of code, I kept them separate.

- [x] copy the header to the correct place
- [x] shortly check if all header includes are correct: remove unused, order alphabetically, put seqan3/std/* headers to the top
- [x] ~~possibly rename the header~~
- [x] deprecate old header with header deprecation macro
- [x] deprecate old header doxygen file section: add [DEPRECATED] at begin of brief and add \deprecated This header is deprecated and will be removed in SeqAn-3.1.0;. Please #include <new header> instead. tag.
- [x] move all related tests, snippets etc to match new folder structure
- [x] run all tests to see if nothing depends on old header anymore
- [x] live happily ever after

### CODE REVIEW

**Review by MitraDarja — APPROVED** [review]

File: include/seqan3/utility/char_operations/all.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/utility/char_operations/predicate.hpp>
+
+/*!\defgroup char_operations builtin_character operations
+ * \brief Provides various operations on character types.
+ * \ingroup core
```
Comment (simonsasse) [inline include/seqan3/utility/char_operations/all.hpp:20]: Should be in group utility right ?
Just stumbled upon this while searching for some inspiration 🧐
```suggestion
 * \ingroup utility
```

Comment (smehringer) [reply include/seqan3/utility/char_operations/all.hpp:20]: yes thanks!

**Comment by smehringer** [comment]
@rrahn polite ping :)

File: include/seqan3/argument_parser/argument_parser.hpp [hunk 2]
```diff
@@ -28,8 +28,8 @@
 #include <seqan3/argument_parser/detail/format_parse.hpp>
 #include <seqan3/argument_parser/detail/terminal.hpp>
 #include <seqan3/argument_parser/detail/version_check.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
 #include <seqan3/core/debug_stream/detail/to_string.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
 #include <seqan3/core/detail/test_accessor.hpp>
 #include <seqan3/io/stream/concept.hpp>
```
Comment (rrahn) [inline include/seqan3/argument_parser/argument_parser.hpp:34]: include order

File: include/seqan3/io/alignment_file/format_bam.hpp [hunk 3]
```diff
@@ -21,7 +21,7 @@
 #include <seqan3/alphabet/detail/convert.hpp>
 #include <seqan3/alphabet/nucleotide/sam_dna16.hpp>
 #include <seqan3/core/bit_manipulation.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/alignment_file/format_bam.hpp:24]: include order

File: include/seqan3/io/alignment_file/format_sam.hpp [hunk 4]
```diff
@@ -19,7 +19,7 @@
 #include <string>
 #include <vector>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/alignment_file/format_sam.hpp:22]: include order

File: include/seqan3/io/alignment_file/format_sam_base.hpp [hunk 5]
```diff
@@ -20,7 +20,7 @@
 #include <string>
 #include <vector>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/alignment_file/format_sam_base.hpp:23]: include order

File: include/seqan3/io/alignment_file/sam_tag_dictionary.hpp [hunk 6]
```diff
@@ -15,7 +15,7 @@
 #include <map>
 #include <variant>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/alignment_file/sam_tag_dictionary.hpp:18]: include order

File: include/seqan3/io/sequence_file/format_embl.hpp [hunk 7]
```diff
@@ -18,7 +18,7 @@
 #include <vector>
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/sequence_file/format_embl.hpp:21]: include order

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 8]
```diff
@@ -20,7 +20,7 @@
 #include <seqan3/alphabet/adaptation/char.hpp>
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
 #include <seqan3/alphabet/quality/aliases.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/sequence_file/format_fasta.hpp:23]: include order

File: include/seqan3/io/sequence_file/format_fastq.hpp [hunk 9]
```diff
@@ -20,7 +20,7 @@
 #include <seqan3/alphabet/adaptation/char.hpp>
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
 #include <seqan3/alphabet/quality/aliases.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/sequence_file/format_fastq.hpp:23]: include order

File: include/seqan3/io/sequence_file/format_genbank.hpp [hunk 10]
```diff
@@ -20,7 +20,7 @@
 #include <range/v3/view/chunk.hpp>
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/sequence_file/format_genbank.hpp:23]: include order

File: include/seqan3/io/structure_file/format_vienna.hpp [hunk 11]
```diff
@@ -22,7 +22,7 @@
 
 #include <seqan3/alphabet/adaptation/char.hpp>
 #include <seqan3/alphabet/structure/wuss.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/io/structure_file/format_vienna.hpp:25]: include order

File: include/seqan3/range/views/take_line.hpp [hunk 12]
```diff
@@ -12,7 +12,7 @@
 
 #pragma once
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline include/seqan3/range/views/take_line.hpp:15]: include order

File: test/performance/range/views/view_take_until_benchmark.cpp [hunk 13]
```diff
@@ -13,7 +13,7 @@
 
 #include <benchmark/benchmark.h>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/performance/range/views/view_take_until_benchmark.cpp:16]: include order

File: test/performance/utility/char_operations/char_predicate_benchmark.cpp [hunk 14]
```diff
@@ -11,7 +11,7 @@
 
 #include <benchmark/benchmark.h>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/performance/utility/char_operations/char_predicate_benchmark.cpp:14]: include order

File: test/snippet/range/views/take_until.cpp [hunk 15]
```diff
@@ -1,6 +1,6 @@
 #include <string>
 
-#include <seqan3/core/char_operations/predicate.hpp> // for is_char
+#include <seqan3/utility/char_operations/predicate.hpp> // for is_char
```
Comment (rrahn) [inline test/snippet/range/views/take_until.cpp:3]: include order

File: test/unit/alphabet/nucleotide/dna15_test.cpp [hunk 16]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/dna15.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/dna15_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/dna3bs_test.cpp [hunk 17]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/dna3bs.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/dna3bs_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/dna4_test.cpp [hunk 18]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/dna4_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/dna5_test.cpp [hunk 19]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/dna5_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/rna15_test.cpp [hunk 20]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/rna15.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/rna15_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/rna4_test.cpp [hunk 21]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/rna4.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/rna4_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/rna5_test.cpp [hunk 22]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/rna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/rna5_test.cpp:9]: include order

File: test/unit/alphabet/nucleotide/sam_dna16_test.cpp [hunk 23]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 #include <seqan3/alphabet/nucleotide/sam_dna16.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/alphabet/nucleotide/sam_dna16_test.cpp:9]: include order

File: test/unit/range/views/istreambuf_test.cpp [hunk 24]
```diff
@@ -12,7 +12,7 @@
 #include <seqan3/std/ranges>
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/range/views/istreambuf_test.cpp:15]: include order

File: test/unit/range/views/persist_test.cpp [hunk 25]
```diff
@@ -12,7 +12,7 @@
 #include <range/v3/algorithm/copy.hpp>
 #include <range/v3/view/unique.hpp>
 
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/range/views/persist_test.cpp:15]: include order

File: test/unit/utility/char_operations/char_predicate_test.cpp [hunk 26]
```diff
@@ -14,7 +14,7 @@
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
 #include <seqan3/alphabet/nucleotide/rna5.hpp>
-#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
```
Comment (rrahn) [inline test/unit/utility/char_operations/char_predicate_test.cpp:17]: include order

**Review by rrahn — CHANGES_REQUESTED** [review]

File: test/snippet/range/views/take_until.cpp [hunk 27]
```diff
@@ -1,10 +1,10 @@
+#include <seqan3/std/ranges>                         // for std::views::reverse
 #include <string>
 
-#include <seqan3/core/char_operations/predicate.hpp> // for is_char
 #include <seqan3/core/debug_stream.hpp>              // for debug_stream
 #include <seqan3/range/views/single_pass_input.hpp>  // for views::single_pass_input
 #include <seqan3/range/views/take_until.hpp>         // for views::take_until*
-#include <seqan3/std/ranges>                         // for std::views::reverse
+#include <seqan3/utility/char_operations/predicate.hpp> // for is_char
```
Comment (rrahn) [inline test/snippet/range/views/take_until.cpp:7]: Please, either align them or remove the alignment.

**Review by rrahn — CHANGES_REQUESTED** [review]

**Review by rrahn — APPROVED** [review]

**Pull request merged**